code-challenges.js

```diff
@@ -6,22 +6,84 @@ var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
-
-
-
+//Create a function that takesin a string and returns a coded message.
+//The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+//Create a function that takes in a string
+const crazyCase = (string) => {
+//create an array of substrings through .split in order to have an array to work with
+  let newString = string.split("")
+//  console.log(newString);
+// .map over your newString since its nested inside an array.
+  var mapOver = newString.map(value => {
+//  if a value in my array is strictly equal to "a"
+//return that value is equal to 4
+//  else if a value in my array is strictly equal to "e"
+//return that value is equal to 3
+//  else if a value in my array is strictly equal to "i"
+//return that value is equal to 1
+//  else if a value in my array is strictly equal to "o"
+//return that value is equal to 0
+//  else return my value (value 'as is')
+    if(value === "a"){
+      return value = 4
+    }else if (value === "e"){
+      return value = 3
+    }else if (value === "i"){
+      return value = 1
+    }else if (value === "o"){
+      return value = 0
```

Comment:
consider a data structure to store the mapping from letter to number

Reply:
Also know that if you're using a `return` in the if-blocks, you don't need to assign the number to the `value`. You can literally return the number (e.g.`return 4`) and it'll continue to iterating onto the next term.